libgalois/src/MemoryPolicy.cpp

```diff
@@ -67,61 +74,58 @@ katana::MemoryPolicyPerformance::MemoryPressureHigh(
 
 count_t
 katana::MemoryPolicyPerformance::ReclaimForMemoryPressure(
-    count_t active, count_t standby) const {
+    count_t standby) const {
   MemInfo mem_info;
-  UpdateMemInfo(&mem_info, active, standby);
+  UpdateMemInfo(&mem_info, standby);
+  count_t reclaim = standby;
 
-  if (mem_info.oom_score < 700 ||
-      mem_info.used_ratio < high_used_ratio_threshold()) {
+  if (mem_info.oom_score < 1000 ||
+      mem_info.used_ratio < high_used_ratio_threshold() ||
+      mem_info.available_bytes > 0.1 * physical()) {
     return 0;
   }
-  if (mem_info.oom_score < 850) {
-    LogIt(
-        fmt::format(
-            "reclaim for memory pressure {} GB", katana::ToGB(standby / 4)),
-        &mem_info);
-    return standby / 4;
-  }
+  // TODO (witchel) this might give back memory too quickly
   if (mem_info.oom_score < 1000) {
-    LogIt(
-        fmt::format(
-            "reclaim for memory pressure {} GB", katana::ToGB(standby / 2)),
-        &mem_info);
-    return standby / 2;
+    reclaim = standby / 4;
+  } else if (mem_info.oom_score < 1200) {
+    reclaim = standby / 2;
   }
   LogIt(
-      fmt::format("reclaim for memory pressure {} GB", katana::ToGB(standby)),
+      fmt::format("reclaim for memory pressure {} GB", katana::ToGB(reclaim)),
       &mem_info);
-  return standby;
+  return reclaim;
 }
 
 bool
 katana::MemoryPolicyPerformance::KillSelfForLackOfMemory(
-    count_t active, count_t standby) const {
+    count_t standby) const {
   MemInfo mem_info;
-  UpdateMemInfo(&mem_info, active, standby);
+  UpdateMemInfo(&mem_info, standby);
 
-  if (mem_info.oom_score > 1200 ||
-      mem_info.used_ratio > kill_used_ratio_threshold()) {
+  if ((mem_info.oom_score > kill_self_oom_threshold() ||
+       mem_info.used_ratio > kill_used_ratio_threshold()) &&
```

Comment:
Is there ever a case where the used ratio is greater than the threshold but the OOM score doesn't exceed the threshold? It seems like they should both go hand in hand?

Reply:
Not necessarily.  The OOM score goes up a lot when a process starts allocating a lot of memory, so there is a velocity associated with the score.  But I don't have a detailed understanding of the components that go into the OOM score.  I should just read the kernel code.